tests/test_utils.py

```diff
@@ -764,6 +764,52 @@ def test_status(self, app):
         result = client.simulate_get()
         assert result.status == falcon.HTTP_702
 
+    @pytest.mark.parametrize(
+        'value',
+        (
+            'd\xff\xff\x00',
+            'Can we consider this a long text for the sake of testing',
+            '{"hello": "WORLD!"}',
+        ),
+    )
+    def test_repr_result(self, asgi, value):
+        if asgi:
+            resource = testing.SimpleTestResourceAsync(body=value)
+        else:
+            resource = testing.SimpleTestResource(body=value)
+
+        app = create_app(asgi)
+        app.add_route('/hello', resource)
+        client = falcon.testing.TestClient(app)
+
+        result = client.simulate_get('/hello')
+        captured_resp = resource.captured_resp
+        content = bytes(captured_resp.text[:20], 'utf-8')
+
+        expected_result = 'Result<{}, {}, {}>'.format(
+            captured_resp.status, captured_resp.headers['content-type'], content
+        )
+
+        assert str(result) == expected_result
+
+        result = client.simulate_post('/hello')
+        captured_resp = resource.captured_resp
+        content = bytes(captured_resp.text[:20], 'utf-8')
+
+        expected_result = 'Result<{}, {}, {}>'.format(
+            captured_resp.status, captured_resp.headers['content-type'], content
+        )
+
+        assert str(result) == expected_result
+
+    def test_repr_without_content_type_header(self, asgi):
+        value = b'huh'
+        header = [('Not-content-type', '!!!')]
+        result = falcon.testing.Result([value], falcon.HTTP_200, header)
+
+        expected_result = 'Result<200 OK, , {}>'.format(value)
```

Comment:
I know I'm nitpicking, but OTOH a representation is about the esthetics :smiling_imp:
What if `Content-Type` or content are empty, the `__repr__` would look somewhat awkward -- as illustrated here.

Maybe we could skip the `Content-Type` if it is empty? Those commas are probably not needed either, although I'm open for suggestions there.

My suggestions could be
`Result<200 OK application/json b'{}'>`
`Result<201 Created>`
`Result<503 Service Unavailable b'Database error, dunno\n'>`

One can also ellipsize the text that was truncated, use lenient decoding and present data as string, the possibilities are endless :slightly_smiling_face:

Reply:
this would be nice to have, but can probably be deferred until some later time

Reply:
> this would be nice to have, but can probably be deferred until some later time

So sorry, i missed this message before implementing the suggested syntax, my latest commit implement the suggestions above... my apologies.

Concerning my latest commit that changes the syntax, i couldn't think of a way to implement the representation of `Result.content` without an if statement, any feedback will be much appreciated!